### PR TITLE
Use newuri for unhash

### PR DESCRIPTION
**Pull request by DancingAxolotl**

### CODE REVIEW

File: Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift [hunk 1]
```diff
@@ -221,6 +221,28 @@ internal class CNS: CommonNamingService, NamingService {
         }
     }
 
+    func getDomainName(tokenId: String) throws -> String {
+        do {
+            let registryContract = try self.buildContract(address: self.contracts.registryAddress, type: .registry)
```
Comment (DeRain) [inline Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift:226]: Let's pull registry address from ProxyReader contract

File: Sources/UnstoppableDomainsResolution/ABI/ABICoder.swift [hunk 2]
```diff
@@ -58,6 +71,21 @@ internal class ABICoder {
         return decoded
     }
 
+    public func decodeEventTopics(eventName: String, eventLogTopics: [Data], eventLogData: Data) throws -> [String: Any] {
```
Comment (DeRain) [inline Sources/UnstoppableDomainsResolution/ABI/ABICoder.swift:74]: Please add simple tests for abicoder

**Review by DeRain — CHANGES_REQUESTED** [review]

File: Sources/UnstoppableDomainsResolution/APIRequest.swift [hunk 3]
```diff
@@ -95,3 +96,36 @@ public struct DefaultNetworkingLayer: NetworkingLayer {
...
+public struct MetadataAPIRequest{
+    public init() { }
+
+    public func makeHttpRequest(url: URL,
+                                completion: @escaping(Result<TokenUriMetadata, Error>) -> Void) {
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = "GET"
+        urlRequest.addValue("application/json", forHTTPHeaderField: "Content-Type")
+
+        let dataTask = URLSession.shared.dataTask(with: urlRequest) { data, response, _ in
```
Comment (rommex) [inline Sources/UnstoppableDomainsResolution/APIRequest.swift:109]: Can we use the `networkingLayer` here instead?
It's a good general idea to provide the client an option to implement networking themselves if they want.

Comment (JohnnyJumper) [reply Sources/UnstoppableDomainsResolution/APIRequest.swift:109]: Somehow I got the same commit in uns branch without the implementation of  MetadataApiRequest. I have moved it on networking layer and used uns one. Even though it is possible to use different networking layers on uns and zns.in #45

File: Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift (hunk 1 above)
Comment (JohnnyJumper) [reply Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift:226]: I believe this is the same as tokenOwnedBy [from master](https://github.com/unstoppabledomains/resolution-swift/pull/44)

File: Sources/UnstoppableDomainsResolution/APIRequest.swift (hunk 3 above)
Comment (DancingAxolotl) [reply Sources/UnstoppableDomainsResolution/APIRequest.swift]: Thanks, I've merged it with changes in #45 
I wasn't aware that we provide an option for custom networking.

**Pull request merged**